gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/iterator/IteratorUtils.java

```diff
@@ -260,10 +301,40 @@ public static <S, E> Iterator<E> cast(final Iterator<S> iterator) {
 
     ///////////////
 
-    public static final <S> Iterator<S> filter(final Iterator<S> iterator, final Predicate<S> predicate) {
+    public static <S> Iterator<S> peek(final Iterator<S> iterator, final Consumer<S> function) {
```

Comment:
it's a simple function but would you mind adding tests in `IteratorUtilsTest` for `peek()` since it's a utility. been trying to maintain the coverage levels on this class.

Reply:
yes